File: use-api.py
# Example: How to use your Medical Assessment API with Python

import requests
import json

API_BASE = 'http://localhost:3000'

def use_api():
    print('🚀 Using Medical Assessment API...\n')

    try:
        # 1. Check API health
        print('1. Checking API health...')
        health_response = requests.get(f'{API_BASE}/health')
        health_data = health_response.json()
        print(f'✅ API Status: {health_data["status"]}')
        print(f'📅 Timestamp: {health_data["timestamp"]}')

        # 2. List available files
        print('\n2. Listing available files...')
        files_response = requests.get(f'{API_BASE}/api/files')
        files_data = files_response.json()
        print('📁 Available files:')
        for file in files_data['files']:
            print(f'   - {file["name"]} ({file["type"]})')

        # 3. Read Excel file data
        print('\n3. Reading Excel file data...')
        excel_response = requests.get(f'{API_BASE}/api/excel/FMG%2009.22.2025.xlsx')
        excel_data = excel_response.json()
        
        if excel_data['success']:
            print('📊 Excel file sheets:')
            for sheet_name, data in excel_data['data'].items():
                print(f'   - {sheet_name}: {len(data)} rows')

        # 4. Process assessments (uncomment to run automation)
        print('\n4. Ready to process assessments...')
        print('⚠️  To start processing, uncomment the code below:')
        print('''
        process_response = requests.post(
            f'{API_BASE}/api/process/FMG%2009.22.2025.xlsx',
            headers={'Content-Type': 'application/json'},
            json={'headless': True, 'slowMo': 2000}
        )
        process_data = process_response.json()
        print('🔄 Processing started:', process_data)
        ''')

    except Exception as error:
        print(f'❌ Error: {error}')

if __name__ == '__main__':
    use_api()

File: use-api.js
// Example: How to use your Medical Assessment API

const API_BASE = 'http://localhost:3000';

async function useAPI() {
  console.log('🚀 Using Medical Assessment API...\n');

  try {
    // 1. Check API health
    console.log('1. Checking API health...');
    const healthResponse = await fetch(`${API_BASE}/health`);
    const healthData = await healthResponse.json();
    console.log('✅ API Status:', healthData.status);
    console.log('📅 Timestamp:', healthData.timestamp);

    // 2. List available files
    console.log('\n2. Listing available files...');
    const filesResponse = await fetch(`${API_BASE}/api/files`);
    const filesData = await filesResponse.json();
    console.log('📁 Available files:');
    filesData.files.forEach(file => {
      console.log(`   - ${file.name} (${file.type})`);
    });

    // 3. Read Excel file data
    console.log('\n3. Reading Excel file data...');
    const excelResponse = await fetch(`${API_BASE}/api/excel/FMG%2009.22.2025.xlsx`);
    const excelData = await excelResponse.json();
    
    if (excelData.success) {
      console.log('📊 Excel file sheets:');
      Object.keys(excelData.data).forEach(sheetName => {
        const rowCount = excelData.data[sheetName].length;
        console.log(`   - ${sheetName}: ${rowCount} rows`);
      });
    }

    // 4. Process assessments (uncomment to run automation)
    console.log('\n4. Ready to process assessments...');
    console.log('⚠️  To start processing, uncomment the code below:');
    console.log(`
    const processResponse = await fetch(\`\${API_BASE}/api/process/FMG%2009.22.2025.xlsx\`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ headless: true, slowMo: 2000 })
    });
    const processData = await processResponse.json();
    console.log('🔄 Processing started:', processData);
    `);

  } catch (error) {
    console.error('❌ Error:', error.message);
  }
}

// Run the example
useAPI();




